src/main/java/org/terasology/web/resources/ResourceManager.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.web.resources;
+
+import org.terasology.context.Context;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.event.internal.EventSystem;
+import org.terasology.entitySystem.systems.ComponentSystem;
+import org.terasology.logic.console.Console;
+import org.terasology.network.NetworkSystem;
+import org.terasology.web.io.ActionResult;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class ResourceManager {
+
+    private static final ResourceManager INSTANCE = new ResourceManager();
+    private Map<String, Resource> resources;
+
+    public static ResourceManager getInstance() {
+        return INSTANCE;
+    }
+
+    public void initialize(Context context) {
+        if (resources == null) {
+            resources = new HashMap<>();
+            putResource(new ConsoleResource(context.get(Console.class)));
+            putResource(new OnlinePlayersResource(context.get(NetworkSystem.class)));
+        }
+        registerEventHandlers(context);
+    }
+
+    private void registerEventHandlers(Context context) {
+        EventSystem eventSystem = context.get(EntityManager.class).getEventSystem();
+        for (Map.Entry<String, Resource> entry: resources.entrySet()) {
+            Resource resource = entry.getValue();
+            if (resource instanceof ComponentSystem) {
```

Comment:
If I remember correctly, all systems that are annotated correctly should receive events. Am I missing something here?